File: Domain/Payment.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Payment
    {
        public Guid Id { get; set; }
        public string ORNumber { get; set; }
        public float Amount { get; set; }
        public string ModeOfPayment { get; set; }
        public DateTime DateOfPayment { get; set; }
        public string CheckNo { get; set; }
        public string BankName { get; set; }
        public string Branch { get; set; }
        public string InPaymentOf { get; set; }
<<<<<<< HEAD
        public float Total {get; set;}
        public Guid TransactionTypeId { get; set; }
        public virtual TransactionType TransactionType { get; set; }
        public Guid TransactionId { get; set; }
        public virtual Transaction Transaction { get; set; }
        public string ReceivedById { get; set; }
        public virtual AppUser AppUser { get; set; }
        // public virtual ICollection<AppUser> ReceivedBy { get; set; }
=======
>>>>>>> 9d19d15bf029e72dd3fe5c7374fc61541e4c14ec
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }

        // public Guid TransactionTypeId { get; set; }
        // public virtual TransactionType TransactionType { get; set; }
        // public Guid TransactionId { get; set; }
        // public virtual Transaction Transaction { get; set; }
        // public string ReceivedById { get; set; }
        // public virtual AppUser AppUser { get; set; }

    }
}

File: client-app/src/app/models/payment.ts
export interface IPayment {
  id: string;
  sequenceNo?: string;
  orNumber: string;
  amount: number | any;
  modeOfPayment: string;
  typeOfPayment: string;
  dateOfPayment?: Date;
  checkNo: string;
  bankName: string;
  branch: string;
  createdAt: Date;
}

export interface IPaymentDto {
  id: string;
  sequenceNo?: string;
  arNumber: string;
  name: string;
  propertyName: string;
  contractPrice: number | any;
  balance: number | any;
  amountPaid: number | any;
  type: string;
  mode: string;
  date?: Date;
}

export interface IPaymentFormValues extends Partial<IPayment> {}

export class PaymentFormValues implements IPaymentFormValues {
  id?: string = '';
  sequenceNo: string = '';
  dateOfPayment?: Date = undefined;
  oRNumber: string = '';
  amount: string = '';
  modeOfPayment: string = '';
  typeOfPayment: string = '';
  checkNo: string = '';
  bankName: string = '';
  branch: string = '';
  // DateOfPayment?:  Date = undefined;
  // InPaymentOf: string = '';
  // Total: string = '';
  // TransactionTypeId?: string = undefined;
  // TransactionId?:string = undefined;
  // ReceivedById?: string = undefined;

  constructor(init?: IPaymentFormValues) {
    Object.assign(this, init);
  }
}

export interface IPaymentFormValues {
  id?: string;
  transactionSequenceNo?: string;
  dateOfPayment?: Date;
  oRNumber: string;
  amount: string;
  typeOfPayment: string;
  modeOfPayment: string;
  checkNo: string;
  bankName: string;
  branch: string;
}
